src/AngleSharp.Diffing/HtmlDiffer.cs

```diff
@@ -24,7 +24,12 @@ public class HtmlDiffer
         public HtmlDiffer(IDiffingStrategy diffingStrategy)
         {
             _diffingStrategy = diffingStrategy ?? throw new ArgumentNullException(nameof(diffingStrategy));
-            var config = Configuration.Default.WithCss();
+
+            // Create a custom config with a parser to allow access to the source reference from the AST.
+            var config = Configuration.Default
+                .WithCss()
+                .With<IHtmlParser>(ctx => new HtmlParser(new HtmlParserOptions { IsKeepingSourceReferences = true, IsScripting = ctx?.IsScripting() ?? false }, ctx));
```

Comment:
@FlorianRappl is there any implications/negative impact of always having `IsKeepingSourceReferences` set to true?

If yes, we may need to make that configurable.

Reply:
There are only performance implications afaik. Because it keeps a reference to this node.

Reply:
The memory consumption will definitely be higher. I'd make it configurable (default to `true`, but if somebody runs into issues it could be disabled).